cmd/kube-controller-manager/app/core.go

```diff
@@ -71,9 +71,23 @@ func startServiceController(ctx ControllerContext) (bool, error) {
 }
 
 func startNodeController(ctx ControllerContext) (bool, error) {
-	_, clusterCIDR, err := net.ParseCIDR(ctx.Options.ClusterCIDR)
-	if err != nil {
-		glog.Warningf("Unsuccessful parsing of cluster CIDR %v: %v", ctx.Options.ClusterCIDR, err)
+	var clusterCIDRs []*net.IPNet
+	for _, cidr := range ctx.Options.ClusterCIDRs {
+		_, cc, err := net.ParseCIDR(cidr)
+		if err != nil {
+			glog.Warningf("Unsuccessful parsing of cluster CIDR %v: %v", ctx.Options.ClusterCIDR[0], err)
+		}
+		clusterCIDRs = append(clusterCIDRs, cc)
+	}
+	NodeCIDRMaskSizes := ctx.Options.NodeCIDRMaskSizes
+	// If the new style was not found, check the old.
+	if len(clusterCIDRs) == 0 {
+		_, cc, err := net.ParseCIDR(ctx.Options.ClusterCIDR)
+		if err != nil {
+			glog.Warningf("Unsuccessful parsing of cluster CIDR %v: %v", ctx.Options.ClusterCIDR, err)
+		}
+		clusterCIDRs = append(clusterCIDRs, cc)
+		NodeCIDRMaskSizes = append(NodeCIDRMaskSizes, int(ctx.Options.NodeCIDRMaskSize))
```

Comment:
I changed it to accept only the old or the new version, but not both.
I can look at adding UT's for these cases also.